docs/dependencies.md

```diff
@@ -157,14 +157,14 @@ highlighting.  The system is largely managed by the code in
   the versions of our indirect dependencies files with
   `tools/update-locked-requirements` (powered by `pip-compile`).  What
   this means is that we have some "source" requirements files, like
-  `requirements/common.txt`, that declare the packages that Zulip
+  `requirements/common.in`, that declare the packages that Zulip
   depends on directly.  Those packages have their own recursive
   dependencies.  When adding or removing a dependency from Zulip, one
-  simply edits the appropriate "source" requirements files, and then
-  runs `tools/update-locked-requirements`.  That tool will use `pip
-  compile` to generate the `prod_lock.txt` and `dev_lock.txt` files
-  that explicitly declare versions of all of Zulip's recursive
-  dependencies.  For indirect dependencies (i.e. dependencies not
+  simply edits the appropriate "source" requirements files(files with
+  ending in `.in`), and then runs `tools/update-locked-requirements`.
+  That tool will use `pip compile` to generate the `prod_lock.txt` and
+  `dev_lock.txt` files that explicitly declare versions of all of Zulip's
+  recursive dependencies.  For indirect dependencies (i.e. dependencies not
```

Comment:
I think we might need to move this section up next to the other one to make things less confusing.

I'm also thinking maybe `mypy.txt` and `docs.txt` don't need to be renamed to have the `.in` extension; it might be clearer that way (`.txt` denoting files that might be directly installed).

Reply:
IMHO not renaming mypy.txt and docs.txt would be more confusing.

Reply:
Well, my thought is that a `.txt` file is one that a user might directly install.  It feels wrong to have our docs recommend e.g. `pip install -r requirements/mypy.in`.  

I suppose that we could move the `mypy.txt` and `docs.txt` into the `update-locked-requirements` framework; the main reason I was thinking of not doing that was perf, but maybe it doesn't matter.

Reply:
>I suppose that we could move the mypy.txt and docs.txt into the update-locked-requirements framework; the main reason I was thinking of not doing that was perf, but maybe it doesn't matter.

Sorry I missed your reply here. We already include mypy.txt and docs.txt in `update-locked-requirements` framework, after doing any changes to mypy.txt or docs.txt, one needs to run `update-locked-requirements`. Is that what you were referring to as including them in the framework?

Reply:
No, I mean treating them as outputs from `update-locked-requirements`; i.e. having a `mypy.in` AND a `mypy.txt` (that is generated from `mypy.in`)

Reply:
Yeah we could do that and doing so would not cause any perf problem as long as we don't install mypy from a particular commit, otherwise it would slow down the tool a lot.

Reply:
Yeah, I like that plan.